internal/provider/resource_connect_artifact.go

```diff
@@ -337,10 +342,10 @@ func connectArtifactProvisionStatus(ctx context.Context, c *Client, environmentI
 		phase := artifact.Status.GetPhase()
 		tflog.Debug(ctx, fmt.Sprintf("Waiting for Connect Artifact %q provisioning status to become %q: current status is %q", artifactId, stateProvisioned, phase), map[string]interface{}{connectArtifactLoggingKey: artifactId})
 
-		if phase == stateProvisioning || phase == stateProvisioned || phase == stateReady {
+		if phase == stateProcessing || phase == stateProvisioning || phase == stateProvisioned || phase == stateReady {
 			return artifact, phase, nil
-		} else if phase == stateFailed {
-			return nil, stateFailed, fmt.Errorf("connect artifact %q provisioning status is %q", artifactId, stateFailed)
+		} else if phase == stateWaitingForProcessing || phase == stateFailed {
```

Comment:
To avoid confusion for anyone maintaining this down the line, I think we should have a short comment explaining that WaitingForProcessing is intended to be a final failed state.

Reply:
That's a great idea! I went ahead and applied the suggestion.

Reply:
update: WAITING_FOR_PROCESSING and PROCESSING should be treat the same way.